core/src/nodes/listeners.rs

```diff
@@ -23,9 +23,58 @@ use std::fmt;
 use void::Void;
 use {Multiaddr, Transport};
 
-/// Implementation of `Stream` that handles listeners.
+/// Implementation of `futures::Stream` that allows listening on multiaddresses.
 ///
-/// The stream cannot produce errors.
+/// To start using a `ListenersStream`, create one with `new` by passing an implementation of
+/// `Transport`. This `Transport` will be used to start listening, therefore you want to pass
+/// a `Transport` that supports the protocols you wish you listen on.
+///
+/// Then, call `ListenerStream::listen_on` as many times as you want to start the actual act of
+/// listening.
```

Comment:
Why would I want to call `listen_on` multiple times? I.e. is there a use case where this is necessary?

Reply:
Well, to listen on multiple different addresses.

Reply:
Right, then I'd suggest: "Then, call `ListenersStream::listen_on` for all addresses you want to start listening on."